cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * XBlockListView shows list of XBlocks in a particular category(section, subsection, vertical etc).
+ */
+define([
+    'jquery', 'backbone', 'underscore', 'gettext',
+    'edx-ui-toolkit/js/utils/html-utils',
+    'edx-ui-toolkit/js/utils/string-utils',
+    'js/views/utils/xblock_utils',
+    'text!templates/move-xblock-list.underscore'
+],
+function($, Backbone, _, gettext, HtmlUtils, StringUtils, XBlockUtils, MoveXBlockListViewTemplate) {
+    'use strict';
+
+    var XBlockListView = Backbone.View.extend({
+        el: '.xblock-list-container',
+
+        // parent info of currently displayed childs
+        parent_info: {},
+        // child info of currently displayed child XBlocks
+        childs_info: {},
+        // list of visited parent XBlocks, needed for backward navigation
+        visitedAncestors: null,
+
+        // parent to child relation map
+        categoryRelationMap: {
+            course: 'section',
+            section: 'subsection',
+            subsection: 'unit',
+            unit: 'component'
+        },
+
+        categoriesText: {
+            section: gettext('Sections'),
+            subsection: gettext('Subsections'),
+            unit: gettext('Units'),
+            component: gettext('Components')
+        },
+
+        events: {
+            'click .button-forward': 'renderChilds'
+        },
+
+        initialize: function(options) {
+            this.visitedAncestors = [];
+            this.template = HtmlUtils.template(MoveXBlockListViewTemplate);
+            this.ancestorInfo = options.ancestorInfo;
+            this.listenTo(Backbone, 'move:backButtonPressed', this.handleBackButtonPress);
+            this.listenTo(Backbone, 'move:breadcrumbButtonPressed', this.handleBreadcrumbButtonPress);
+            this.renderXBlockInfo();
+        },
+
+        render: function() {
+            HtmlUtils.setHtml(
+                this.$el,
+                this.template(
+                    {
+                        xblocks: this.childs_info.childs,
+                        noChildText: this.getNoChildText(),
+                        categoryText: this.getCategoryText(),
+                        XBlocksCategory: this.childs_info.category,
+                        forwardButtonSRText: this.getForwardButtonSRText(),
+                        currentLocationIndex: this.getCurrentLocationIndex()
+                    }
+                )
+            );
+            Backbone.trigger('move:childsInfoRendered', this.breadcrumbInfo());
+            return this;
+        },
+
+        renderChilds: function(event) {
+            this.renderXBlockInfo(
+                'forward',
+                $(event.target).closest('.xblock-item').data('itemIndex')
+            );
+        },
+
+        renderParent: function(newParentIndex) {
+            this.renderXBlockInfo('backward', newParentIndex);
+        },
```

Comment:
I think we don't need this method, we can call `this.renderXBlockInfo('backward', newParentIndex);` directly from other places.